scrunch/helpers.py

```diff
@@ -6,6 +6,26 @@
 else:
     from urllib.parse import urljoin
 
+
+class ReadOnly(object):
+    """
+    class for protecting undesired writes to attributes
+    """
+    def __init__(self, resource):
+        object.__setattr__(self, "resource", resource)
+        object.__setattr__(self, "session", resource.session)
```

Comment:
you're right. Will update and change the code where the session was accessed directly.